JASP-Engine/JASP/R/regressionloglinear.R

```diff
@@ -15,559 +15,457 @@
+        observations.amount  = "< 1",
+        exitAnalysisIfErrors = TRUE)
+  }
+} 
+
+# Compute results ----
+.regLogLinComputeModel <- function(jaspResults, dataset, options) {
+  if (!is.null(jaspResults[["Model"]])) 
+    return(jaspResults[["Model"]]$object)
+  
+  modelTerms <- unlist(options$modelTerms)
+  if (options$counts == "")
+    dataset <- plyr::count(dataset)
+  
+  loglm.model <- list()
+  empty.model <- list(loglm.fit = NULL, variables = NULL)
+  
+  if (options$counts == "")
+    dependent.variable <- "freq"
+  else 
+    dependent.variable <- unlist(options$counts)
+  
+  if (length(options$modelTerms) > 0) {
+    
+    variables.in.model <- NULL
+    variables.in.model.base64 <- NULL
+    
+    for (i in seq_along(options$modelTerms)) {
+      
+      components <- options$modelTerms[[i]]$components
+      
+      if (length(components) == 1) {
+        variables.in.model <- c(variables.in.model, components[[1]])
+        variables.in.model.base64 <- c(variables.in.model.base64, .v(components[[1]]))
+      } else {
+        components.unlisted <- unlist(components)
+        term.base64         <- paste0(.v(components.unlisted), collapse = ":")
+        term                <- paste0(components.unlisted, collapse = ":")
+        variables.in.model  <- c(variables.in.model, term)
+        variables.in.model.base64 <- c(variables.in.model.base64, term.base64)
+      }
+    }
+    
+    dependent.base64 <- .v(dependent.variable)
+    independent.base64 <- variables.in.model.base64
+    variables.in.model <- variables.in.model[ variables.in.model != ""]
+    variables.in.model.copy <- variables.in.model
+    
+    if (length(variables.in.model) > 0 )
+      model.definition <- paste(dependent.base64, "~", 
+                                paste(independent.base64, collapse = "+"))
+    else 
+      model.definition <- NULL #this model has no parameters
+    
+    if (!is.null(model.definition)) {
+      
+      model.formula <- as.formula(model.definition)
+      
+      if (options$counts == "")
+        names(dataset)[names(dataset) == "freq"] <- dependent.base64
+      
+      loglm.fit <- try(stats::glm(model.formula, family = poisson(), 
+                                  data = dataset), silent = TRUE)
+      
+      if (inherits(loglm.fit, "glm"))
+        loglm.model <- list(loglm.fit = loglm.fit, 
+                            variables = variables.in.model)
+    } else 
+      loglm.model <- list(loglm.fit = NULL, 
+                          variables = variables.in.model)
+  } else 
+    loglm.model <- empty.model
+  
+  jaspResults[["Model"]] <- createJaspState(loglm.model)
+  jaspResults[["Model"]]$dependOn(c("counts","modelTerms"))
+  
+  return(loglm.model)
+}
 
-	} else {
+.regLogLinAnovaResults <- function(jaspResults, dataset, options){
+  results <- list()
+  #for empty elements in computed tables
+  empty.line  <- .regloglinAnovaLine(options, char = "",  name = "")  
```

Comment:
This is already handled automatically, right? If no data is added we automatically get a row of dots